### PR TITLE
Make building Python bindings optional

### PR DESCRIPTION
**Pull request by fzakaria**
Many developers are using the ci_build_* scripts to build LLVM and StableHLO.

A previous PR #1923 made the CI system by default building python bindings however there is an undocumented step of installing Python dependencies such as pybind.

To simplify local development, make the option configurable with the default to OFF so that the ci_build_* actions work without any additional setup.

The CI system itself (GitHub actions) set the appropriate variables to build the Python bindings.

### CODE REVIEW

File: build_tools/github_actions/ci_build_cmake.sh [hunk 1]
```diff
@@ -27,21 +27,25 @@ fi
...
   -DCMAKE_CXX_COMPILER_LAUNCHER=ccache \
   -DCMAKE_C_COMPILER=clang \
   -DCMAKE_C_COMPILER_LAUNCHER=ccache \
   -DSTABLEHLO_ENABLE_STRICT_BUILD=ON \
-  -DSTABLEHLO_ENABLE_BINDINGS_PYTHON=ON
+  -DSTABLEHLO_ENABLE_BINDINGS_PYTHON="${STABLEHLO_ENABLE_BINDINGS_PYTHON}"
 
 # Build and Test StableHLO
-cd "$STABLEHLO_BUILD_DIR"
+cd "$STABLEHLO_BUILD_DIR" || exit
```
Comment (fzakaria) [inline build_tools/github_actions/ci_build_cmake.sh:50]: This was an error ShellCheck (https://www.shellcheck.net/) gave me in my editor.
Could be yet another lint to apply :)

Comment (mlevesquedion) [reply build_tools/github_actions/ci_build_cmake.sh:50]: Neat! I personally like setting `set -o errexit` (or `set -e` for short) at the very top of the script. In this case if `cmake` fails we probably don't want to call `ninja` either.

While we're on this, I also prefer `cmake --build "$STABLEHLO_BUILD_DIR" --target check-stablehlo-ci`, but calling `cd`ing and calling `ninja` is also fine.

**Review by mlevesquedion — APPROVED** [review]
Nice, thank you!

**Comment by fzakaria** [comment]
image

Wanted to confirm this still does the right thing.

File: build_tools/github_actions/ci_build_cmake.sh (hunk 1 above)
Comment (fzakaria) [reply build_tools/github_actions/ci_build_cmake.sh:50]: Yes we should probably apply best practices for shellcheck;
I also like the `cmake --build` approach but it's more "new"

**Pull request merged**